src/main/java/ladder/controller/LadderController.java

```diff
@@ -9,22 +11,50 @@
 
 public class LadderController {
 
-    private Names names;
-    private Rows rows;
+    private static final String ALL_PLAYER = "all";
 
     public void run() {
-        names = Repeater.repeatIfError(this::inputNames, ResultView::printErrorMessage);
-        rows = Repeater.repeatIfError(this::inputLines, ResultView::printErrorMessage);
-        rows.generateLegsOfLines(new RandomGenerator());
-        ResultView.printResult(names.toDto(), rows.toDto());
+        LadderGame ladderGame = initializeGame();
+        ladderGame.play();
+        ResultView.printLadder(ladderGame.getNames(), ladderGame.getRows(), ladderGame.getItemNames());
+        repeat(() -> printResult(ladderGame));
     }
 
-    private Names inputNames() {
-        return new Names(InputView.inputPlayerNames());
+    private LadderGame initializeGame() {
+        Players players = Repeater.repeatIfError(this::inputPlayerNames, ResultView::printErrorMessage);
+        Items items = Repeater.repeatIfError(() -> inputItemNames(players.getCount()), ResultView::printErrorMessage);
+        Rows rows = Repeater.repeatIfError(() -> inputLines(players), ResultView::printErrorMessage);
+
+        return new LadderGame(rows, items, players);
+    }
+
+    private Players inputPlayerNames() {
+        return new Players(InputView.inputPlayerNames());
+    }
+
+    private Items inputItemNames(int playerCount) {
+        return new Items(InputView.inputItemNames(), playerCount);
+    }
+
+    private Rows inputLines(Players players) {
+        int intervalCount = players.getCount() - 1;
+        return new Rows(InputView.inputLadderHeight(), intervalCount, new RandomGenerator());
+    }
+
+    private void printResult(LadderGame ladderGame) {
+        String targetPlayerName = InputView.wantToSeeResult();
+        if (targetPlayerName.equals(ALL_PLAYER)) {
```

Comment:
```suggestion
        if (ALL_PLAYER.equals(targetPlayerName)) {
```
이렇게 비교하면 null pointer exception을 방지할 수 있습니다ㅎㅎ

Reply:
확실히 있는 부분에 대해서 체크가 가능하기에 안정적으로 검증이 되겠네요 감사합니다